packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/EnrichmentService.java

```diff
@@ -0,0 +1,231 @@
+
+			// Get the Document
+			DocumentAsset document = documentService
+				.getAsset(documentId, permission)
+				.orElseThrow(() -> {
+					final String errorString = String.format("Document %s not found", documentId);
+					log.warn(errorString);
+					return new IOException(errorString);
+				});
+
+			// Make sure there is text in the document
+			if (document.getText() == null || document.getText().isBlank()) {
+				final String errorString = String.format("Document %s has no extracted text", documentId);
+				log.warn(errorString);
+				throw new IOException(errorString);
+			}
+
+			notificationInterface.sendMessage("Document text found.");
+
+			// Get the model
+			Model model = modelService
+				.getAsset(modelId, permission)
+				.orElseThrow(() -> {
+					final String errorString = String.format("Model %s not found", modelId);
+					log.warn(errorString);
+					return new IOException(errorString);
+				});
+
+			notificationInterface.sendMessage("Model found.");
+
+			// Stripping the metadata before it's sent since it can cause GoLLM to fail with massive inputs
+			model.setMetadata(new ModelMetadata());
+			notificationInterface.sendMessage("Model metadata stripped.");
+
+			// Create the tasks
+			final TaskRequest enrichAmrRequest = TaskUtilities.getEnrichAMRTaskRequest(
+				currentUserService.get().getId(),
+				document,
+				model,
+				projectId
+			);
+
+			final TaskRequest modelCardRequest = TaskUtilities.getModelCardTask(
+				currentUserService.get().getId(),
+				document,
+				model,
+				projectId
+			);
+
+			final TaskRequest taskRequest = new CompoundTask(enrichAmrRequest, modelCardRequest);
+
+			try {
+				taskService.runTask(TaskService.TaskMode.SYNC, taskRequest);
+			} catch (final Exception e) {
+				final String errorString = String.format("Task failed: %s", e);
+				log.warn(errorString);
+				throw new IOException(errorString);
+			}
+
+			notificationInterface.sendMessage("Model enrichment and model card complete.");
+
+			// Get the enriched model
+			final Model enrichedModel = modelService
+				.getAsset(modelId, permission)
+				.orElseThrow(() -> {
+					final String errorString = String.format("Enriched Model %s not found", modelId);
+					log.warn(errorString);
+					return new IOException(errorString);
+				});
+
+			notificationInterface.sendMessage("Updating model grounding...");
+
+			// Update State Grounding
+			if (enrichedModel.isRegnet()) {
+				final List<RegNetVertex> vertices = enrichedModel.getVerticies();
+				vertices.forEach(vertex -> {
+					if (vertex == null) {
+						vertex = new RegNetVertex();
+					}
+					TaskUtilities.performDKGSearchAndSetGrounding(dkgService, vertex);
+				});
+				enrichedModel.setVerticies(vertices);
+			} else {
+				final List<State> states = enrichedModel.getStates();
+				states.forEach(state -> {
+					if (state == null) {
+						state = new State();
+					}
```

Comment:
I don't think this if-section does anything, what are we trying to do?
If state is null, `new State()` does not change the element in the array/list ... 

```
import java.util.List;
import java.util.ArrayList;

public class JavaRef {

    public static void main(String args[]) {
       ArrayList list = new ArrayList();
       list.add(1);
       list.add(null);
       list.add(2);
       list.forEach(elem -> {
            if (elem == null) {
                System.out.println("reassign null");
                elem = Long.valueOf(99);
            }
       });
       System.out.println(list);
    }
}
```

Running this the `null` is unchanged

```
reassign null
[1, null, 2]
```

Reply:
Good point, I should not have copy/paste code. Fixed.